mirai-core-api/src/commonMain/kotlin/utils/SecretsProtection.kt

```diff
@@ -0,0 +1,198 @@
+            // @context: buffer.remaining >= buffer.size
+
+            // 返回存储 data 的数据的 DirectByteBuffer
+            //   ByteBuffer.slice(): DirectByteBuffer[start=pos, end=limit]
+            val mirror = buffer.buffer.let { buffer0 ->
+                buffer0.limit = buffer0.pos + size
+                buffer0.slice()
+            }
+            // 原始缓冲区复位
+            buffer.buffer.let { buf -> buf.limit = buf.capacity() }
+            buffer.buffer.pos += size
+
+            // 此缓冲区已无可用空间
+            if (!buffer.buffer.hasRemaining()) {
+                pool.remove(buffer)
+            }
+
+            return mirror
+        }
+
+        pool.forEach bufferLoop@{ buffer ->
+
+            val bufferRemaining = buffer.buffer.remaining
+            if (bufferRemaining >= size) {
+                if (buffer.lock.tryLock()) {
+                    try {
+                        if (buffer.buffer.hasRemaining(size)) {
+                            NativeBufferWithLock.unusedHitUpdater.getAndIncrement(buffer)
+                            return putInto(buffer)
+                        }
+                    } finally {
+                        buffer.lock.unlock()
+                    }
+                }
+            }
+            NativeBufferWithLock.unusedHitUpdater.getAndDecrement(buffer)
+
+            // OOM Avoid
+            if (bufferRemaining <= lowRemainingThreshold) {
+                NativeBufferWithLock.lowRemainingHitUpdater.getAndDecrement(buffer)
+                if (buffer.lowRemainingHit >= lowRemainingHitThreshold) {
+                    pool.remove(buffer)
+                }
+            }
+            if (buffer.unusedHit >= 20) {
+                pool.remove(buffer)
+            }
+
+        }
+        val newBuffer = NativeBufferWithLock(ByteBuffer.allocateDirect(bufferSize))
+        val rsp = putInto(newBuffer)
+        if (newBuffer.buffer.hasRemaining()) {
+            pool.add(newBuffer)
+        }
+        return rsp
+    }
+
+    @JvmStatic
+    public fun escape(data: ByteArray): ByteBuffer {
+        return allocate(data.size).also {
+            it.put(data)
+            it.pos = 0
+        }
+    }
+
+    @JvmInline
+    @Serializable(EscapedStringSerializer::class)
+    public value class EscapedString(
+        public val data: ByteBuffer,
+    ) {
+        public val asString: String
+            get() = data.duplicate().readString()
+    }
+
+    @JvmInline
+    @Serializable(EscapedByteBufferSerializer::class)
+    public value class EscapedByteBuffer(
+        public val data: ByteBuffer,
+    )
+
+    public object EscapedStringSerializer : KSerializer<EscapedString> by String.serializer().map(
+        String.serializer().descriptor.copy("EscapedString"),
+        deserialize = { EscapedString(escape(it.toByteArray())) },
+        serialize = { it.data.duplicate().readString() }
+    )
+
+    public object EscapedByteBufferSerializer : KSerializer<EscapedByteBuffer> by serializer<ByteArray>().map(
```

Comment:
```suggestion
    public object EscapedByteBufferSerializer : KSerializer<EscapedByteBuffer> by ByteArraySerializer().map(
```
`serializer<ByteArray>()` 不是编译期处理的, 且依赖反射